integration/spark/src/main/scala/org/apache/spark/sql/execution/command/mutation/merge/CarbonMergeDataSetCommand.scala

```diff
@@ -212,17 +212,20 @@ case class CarbonMergeDataSetCommand(
     }
 
     val dataFrame = loadDF.select(tableCols.map(col): _*)
-    CarbonInsertIntoCommand(databaseNameOp = Some(carbonTable.getDatabaseName),
-      tableName = carbonTable.getTableName,
-      options = Map("fileheader" -> header),
-      isOverwriteTable = false,
-      dataFrame.queryExecution.logical,
-      carbonTable.getTableInfo,
-      Map.empty,
-      Map.empty,
-      new OperationContext,
-      updateTableModel
-    ).run(sparkSession)
+    if (mergeMatches.matchList.flatMap(_.getActions).exists(p => p.isInstanceOf[InsertAction]) ||
```

Comment:
Can we move this and the check in CarbonProjectForUpdateCommand.scala to a common place in InsertIntoCommand?

Reply:
handled, please check